cmd/ec2.go

```diff
@@ -15,21 +20,57 @@ var ec2Cmd = &cobra.Command{
 	Short: "Print ec2 related information",
 	Long: `For example,
 		awsctl get ec2`,
-	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("ec2 called")
-	},
+	Run: getEC2command,
 }
 
-func init() {
-	getCmd.AddCommand(ec2Cmd)
+// type EC2DesAPI interface {
+// 	DescribeInstances(ctx context.Context,
+// 		params *ec2.DescribeInstancesInput,
+// 		optFns ...func(*ec2.Options)) (*ec2.DescribeInstancesOutput, error)
+// }
+
+// // Returns info about ec2 instance
+// func GetEc2Info(ctx context.Context, input *ec2.DescribeInstancesInput, api EC2DesAPI) (*ec2.DescribeInstancesOutput, error) {
+
+// 	return api.DescribeInstances(ctx, input)
+
+// }
+
+func getEC2command(cmd *cobra.Command, args []string) {
+
+	ctx := context.TODO() // Unclear Context
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		log.Fatal("Config Error Occured", err)
+	}
 
-	// Here you will define your flags and configuration settings.
+	client := ec2.NewFromConfig(cfg)
+	
 
-	// Cobra supports Persistent Flags which will work for this command
-	// and all subcommands, e.g.:
-	// ec2Cmd.PersistentFlags().String("foo", "", "A help for foo")
+	input := &ec2.DescribeInstancesInput{}
 
-	// Cobra supports local flags which will only run when this command
-	// is called directly, e.g.:
-	// ec2Cmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+	info, err := pkg.GetEc2Info(ctx, input, client)
+	if err != nil {
+		log.Fatal("Error Occured while retrieving Information", err)
+	}
+
+	//Print Instance Details
+	instanceName:=""
+	fmt.Println("NAME \t INSTANCE_ID \t    INSTANCE_TYPE \t PRIVATE IP")
+	for _, r := range info.Reservations {
+		
+		for _, i := range r.Instances {
+			for _,j:= range i.Tags{
+				if *j.Key == "Name"{
+					instanceName=*j.Value
```

Comment:
Instead of `*j.Value` you can use `aws.ToString`

Reply:
same applies to other places as well

Reply:
If nil is return it will create a runtime error. And aws.String require *string if I want to print  nil value which is *& it won't accept it

Reply:
![Screenshot from 2023-02-26 18-12-26](https://user-images.githubusercontent.com/72307121/221411128-7e4f7ccc-d29a-4a87-bf72-a406da958953.png)

Reply:
And If I used aws.String 
![image](https://user-images.githubusercontent.com/72307121/221411171-46984146-2d18-4da7-9b33-3e248514b2c9.png)

Reply:
`aws.Tostring` and not `aws.String`